### DISCUSSION TITLE
TS class for Go struct conversion is incorrect

### DISCUSSION THREAD

**Discussion by josexy** [wailsapp/wails#2646] [Q&A]
I created several Go structs with fields using struct slice pointers. 

```go
type Config struct {
	Addr string        `json:"addr"`
	Port int           `json:"port"`
	Http []*HttpOption `json:"http"`
	Ws   []*WsOption   `json:"ws"`
}

type HttpOption struct {
	Others *Others `json:"others"`
}

type WsOption struct {
	Others *Others `json:"others"`
}

type Others struct {
	Msg string `json:"msg"`
}
```

Now I want to generate Typescript class code, but the `convertValues` in the generated TS class constructor seems to be incorrect (the second parameter is `null`).

```ts
export namespace main {
	export class WsOption {
	    // Go type: Others
	    others?: any;
        // ...
	    constructor(source: any = {}) {
	        if ('string' === typeof source) source = JSON.parse(source);
	        this.others = this.convertValues(source["others"], null);
	    }
        // ...
	}
	export class Others {
	    msg: string;
        // ...
	}
	export class HttpOption {
	    // Go type: Others
	    others?: any;
        // ...
	    constructor(source: any = {}) {
	        if ('string' === typeof source) source = JSON.parse(source);
	        this.others = this.convertValues(source["others"], null);
	    }
        // ...
	}
	export class Config {
	    addr: string;
	    port: number;
	    http: HttpOption[];
	    ws: WsOption[];
        // ...
	    constructor(source: any = {}) {
	        if ('string' === typeof source) source = JSON.parse(source);
	        this.addr = source["addr"];
	        this.port = source["port"];
	        this.http = this.convertValues(source["http"], HttpOption);
	        this.ws = this.convertValues(source["ws"], WsOption);
	    }
        // ...
	}

}
```

When I change `[]*HttpOption` and `[]*WsOption` to `[]HttpOption` and `[]WsOption`, the generated TS class is correct.

I looked at the wails source code, and the problem seems to be here:

https://github.com/wailsapp/wails/blob/8d7c033a6674e6b3fbf99f545044375fa14106b7/v2/internal/binding/binding.go#L205-L221

**Comment by stffabi** [comment]
Would it be possible for you to test the laster master version? I suspect this is a duplicate of #2489, where a fix has been merged into master recently.

**Reply by josexy** [reply]
Yes, I tested the v2.4.0 version, but the above problems still exist

**Reply by stffabi** [reply]
I meant to test the latest master commit, not the latest release. The fix has not yet been released.

**Reply by josexy** [reply]
Of course, I downloaded it through git clone, and then compiled the latest wails binary program

**Reply by stffabi** [reply]
Did you adjust your `go.mod`?

**Reply by josexy** [reply]
Sure, I use go.work 
```
go 1.20

use (
	.
	../wails/v2
)
```

**Reply by stffabi** [reply]
AFAIK `go.work` does not automatically add a replace or update the dependency of your wails app. So it would be interesting to check if in your `go.mod` the wails dependency points to the master commit?

**Reply by josexy** [reply]
I don't think it has much to do with dependency... :(

**Comment by leaanthony** [comment]
Please test using [this guide](https://wails.io/docs/guides/local-development). Take note of [this](https://wails.io/docs/guides/local-development/#updating-your-project) 👍

**Reply by josexy** [reply]
I've done what you said, but it still doesn't work
